### PR TITLE
Voting

### PR DESCRIPTION
**Pull request by CommanderTso**
- Implements voting both for http and ajax
- Lots of tests!  
- Switches test framework to use PhantomJS / Poltergeist, and moves the db cleaning strategy to truncation
- Updates/additions to seed.rb and various factories
- Cleanup of multiple small bugs / issues

### CODE REVIEW

File: app/models/vote.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  validates :score, numericality: true, inclusion: { in: -1..1, message: "A vote must be -1, 0, or 1" }
+
+  def change_vote(new_vote)
+    self.score = if self.score == new_vote
+                   0
+                 else
+                   new_vote
+                 end
+    self.save
+    self.score
```
Comment (CommanderTso) [inline app/models/vote.rb:16]: Keeping these, since it breaks when I take them out.

File: app/assets/javascripts/vote_ajax.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			method: "GET",
+			url: $(this).attr('ajax_path')
+		});
+
+    request.done(function(data) {
+      if ('new_score' in data) {
+        $('li#review-rating-' + data.review_id).text("Review rating: " + data.new_score);
+      } else {
+        if ($('div.flash_message').text != data.login_error) {
+           $('div.flash_message').text(data.login_error)
```
Comment (CommanderTso) [inline app/assets/javascripts/vote_ajax.js:15]: Fixed the quotes issue.  Keeping the variables named in Ruby style since that's where they originate - I'll just get different hound errors for changing them.

**Comment by CommanderTso** [comment]
Pushing in spite of lowered coverage - we're not exercising the code paths for the JS / AJAX feature for up and down voting, since we can't get those to pass on CI - Codeship has some sort of issue with the tests that cannot be replicated locally.
